.gitignore

```diff
@@ -0,0 +1,8 @@
+Exercises/dep
+*.o
+*.host
+*.cuda
+KokkosCore_config.h
+KokkosCore_config.tmp
+*.a
+
```

Comment:
I am in general against .gitignores if possible ... I want to see what changed and I think devs should be aware of what they push ...